google/cloud/bigtable/benchmarks/mutation_batcher_throughput_benchmark.cc

```diff
@@ -154,52 +194,18 @@ int main(int argc, char* argv[]) {
   for (auto i = options->mutation_count - 1; i != 0; i /= 10) ++key_width;
 
   // Create a new table if one was not supplied
-  auto table_id = options->table_id;
-  if (table_id.empty()) {
-    auto generator = google::cloud::internal::MakeDefaultPRNG();
-    table_id = RandomTableId(generator);
-
-    google::bigtable::admin::v2::CreateTableRequest r;
-    r.set_parent(cbt::InstanceName(options->project_id, options->instance_id));
-    r.set_table_id(table_id);
-    // Provide initial splits to the table
-    for (auto i = 0; i != options->shard_count; ++i) {
-      auto row_index = options->mutation_count * i / options->shard_count;
-      r.add_initial_splits()->set_key(MakeRowString(key_width, row_index));
-    }
-    auto& families = *r.mutable_table()->mutable_column_families();
-    families[options->column_family].mutable_gc_rule()->set_max_num_versions(
-        10);
-
-    std::cout << "# Creating Table\n";
-    auto table = admin.CreateTable(std::move(r));
-    if (!table) {
-      std::cout << table.status() << std::endl;
-      return 1;
-    }
-    std::cout << "#\n";
-  } else {
-    google::bigtable::admin::v2::GetTableRequest r;
-    r.set_name(
-        cbt::TableName(options->project_id, options->instance_id, table_id));
-    r.set_view(google::bigtable::admin::v2::Table_View_NAME_ONLY);
-    auto table = admin.GetTable(std::move(r));
-    if (!table) {
-      std::cout << "Error trying to get Table " << table_id << ":\n"
-                << table.status() << std::endl;
-      return 1;
-    }
-  }
+  auto table_id = CreateTableIfNeeded(admin, *options, key_width);
```

Comment:
I was skeptical about the `readability-function-cognitive-complexity` project, but this is definitely nicer. Thanks.